src/ChurchCRM/Service/DashboardService.php

```diff
@@ -38,42 +18,78 @@ public function getFamilyCount(): array
+            $sInactiveClassificationIds = '-1';
+        }
+        $aInactiveClassificationIds = explode(',', $sInactiveClassificationIds);
+        $aDirRoleChild = explode(',', SystemConfig::getValue('sDirRoleChild'));
+
+        $personCount = 0;
+        $ageStats = [];
+        $classificationStats = [];
+        $genderStats = [];
+        $familyRoleStats = [];
+
+        $people = PersonQuery::Create('per')
+            ->filterByClsId($aInactiveClassificationIds, Criteria::NOT_IN)
             ->useFamilyQuery('fam', 'left join')
             ->filterByDateDeactivated(null)
-            ->endUse()
-            ->count();
+            ->endUse();
 
-        return ['personCount' => $personCount];
-    }
+        foreach ($people as $person) {
+            $personCount++;
 
-    /**
-     * @return array<string, string>
-     */
-    public function getPersonStats(): array
-    {
-        $data = [];
-        $sSQL = <<<SQL
-SELECT
-    lst_OptionName as Classification,
-    count(*) as count
-FROM person_per
-INNER JOIN list_lst ON  per_cls_ID = lst_OptionID
-LEFT JOIN family_fam ON family_fam.fam_ID = person_per.per_fam_ID
-WHERE
-    lst_ID = 1 AND
-    family_fam.fam_DateDeactivated IS NULL
-GROUP BY per_cls_ID, lst_OptionName
-ORDER BY count DESC;
-SQL;
-        $rsClassification = RunQuery($sSQL);
-        while ($row = mysqli_fetch_array($rsClassification)) {
-            $data[$row['Classification']] = $row['count'];
+            $classification = $person->getClassificationName();
+            if (!array_key_exists($classification, $classificationStats)) {
+                $classificationStats[$classification]['count'] = 0;
+                $classificationStats[$classification]['id'] = $person->getClsId();
+            }
+            $classificationStats[$classification]['count']++;
+
+            $gender = $person->getGender();
+            if ($gender != 0 && in_array($person->getFmrId(), $aDirRoleChild)) {
+                $gender = $gender + 2;
+            }
+            if (!array_key_exists($gender, $genderStats)) {
+                $genderStats[$gender] = 0;
+            }
+            $genderStats[$gender]++;
+
+            $numericAge = (int) $person->getNumericAge();
+            if ($numericAge != 0) {
+                if (!array_key_exists($numericAge, $ageStats)) {
+                    $ageStats[$numericAge] = 0;
+                }
+                $ageStats[$numericAge]++;
+            }
+
+            $familyRoleGender = $person->getFamilyRoleName() . ' - ' . $person->getGenderName();
+            if (!array_key_exists($familyRoleGender, $familyRoleStats)) {
+                $familyRoleStats[$familyRoleGender]['count'] = 0;
+                $familyRoleStats[$familyRoleGender]['genderId'] = $person->getGender();
+                $familyRoleStats[$familyRoleGender]['roleId'] = $person->getFmrId();
+            }
+            $familyRoleStats[$familyRoleGender]["count"]++;
```

Comment:
single quotes for consistency

```suggestion
            $familyRoleStats[$familyRoleGender]['count']++;
```

Reply:
Done